packages/dev/addons/src/lottie/maths/matrix.ts

```diff
@@ -0,0 +1,233 @@
+import type { IMatrixLike, IVector2Like } from "core/Maths/math.like";
+import type { Tuple } from "core/types";
+
+/**
+ * Represents a Babylon.js thin version of a Matrix
+ * We are only exposing what we truly need in the scope of
+ * the Lottie Renderer project preventing the dependency on the full
+ * Babylon.js math system.
+ */
+export class ThinMatrix implements IMatrixLike {
+    /**
+     * Prevents global conflicts on update as shader programs are shared.
+     */
+    private static _UPDATE_FLAG_SEED = 0;
+
+    /**
+     * Current update flag used to know whether a Matrix has changed since previous render or not.
+     * This helps identifying if a Matrix needs to be
+     * uploaded to a shader program.
+     */
+    public updateFlag = -1;

```

Comment:
Is this only intended to be mutated from within the class? If so, private + public accessor?

Reply:
The implementation of thinMath.matrix.ts also uses a public field, leaving it as is.